CRM/Upgrade/Incremental/php/FourSeven.php

```diff
@@ -208,6 +208,7 @@ public function upgrade_4_7_7($rev) {
   public function upgrade_4_7_8($rev) {
     $this->addTask(ts('Upgrade DB to %1: SQL', array(1 => $rev)), 'runSql', $rev);
     $this->addTask('Upgrade mailing foreign key constraints', 'upgradeMailingFKs');
+    $this->addTask('Add Smartgroup refresh options', 'addSmartGroupRefreshOptions');
```

Comment:
So I wanted to tweak the descriptions for the options and realized they would only be populated for upgrades (not for new installations). And the Setting API doesn't support `pseudoconstant.optiongroup`. It only supports `pseudoconstant.callback`.

`pseudoconstant.callback`  is actually better in this case. There's no reason for an end-user to manipulate the list of possible smart-group-cache-modes. (There could be cases for extensions, but it's better to have a hook like `CRM_Utils_Hook::fieldOptions` which is less dependent on how the options are stored/generated.)

Working on an update...